wannierberri/__sym_wann.py

```diff
@@ -0,0 +1,554 @@
+                                            self.H_select[rot_map[atom_a],rot_map[atom_b]],new_Rvec_index,:].dot(np.transpose(
+                                            self.symmetry['rotations'][rot]) )
+                            else:
+                                if new_Rvec in tmp_R_list:
+                                    pass
+                                else:
+                                    tmp_R_list.append(new_Rvec)
+
+                for atom_a in range(self.num_wann_atom):
+                    for atom_b in range(self.num_wann_atom):
+                        '''
+                        H_ab_sym = P_dagger_a dot H_ab dot P_b
+                        H_ab_sym_T = ul dot H_ab_sym.conj() dot ur
+                        '''
+                        tmp = np.dot(np.dot(p_map_dagger[atom_a],Ham_all[:,atom_a,atom_b]),p_map[atom_b])
+                        if sym_only: 
+                            Ham_res += tmp.transpose(0,2,1)
+                        if sym_T:
+                            tmp_T = self.ul.dot(tmp.transpose(1,0,2)).dot(self.ur).conj()
+                            Ham_res += tmp_T.transpose(0,2,1)
+
+                        for X in self.matrix_list:  # vector matrix
+                            if self.matrix_bool[X]:
+                                vars()[X+'_shift'] = vars()[X+'_all'].transpose(0,1,2,5,3,4)
+                                tmpX= np.dot(np.dot(p_map_dagger[atom_a],vars()[X+'_shift'][:,atom_a,atom_b]),p_map[atom_b])
+                                if sym_only:
+                                    vars()[X+'_res'] += tmpX.transpose(0,3,1,2)
+                                if sym_T:
+                                    tmpX_T = self.ul.dot(tmpX.transpose(1,2,0,3)).dot(self.ur).conj()
+                                    vars()[X+'_res'] += tmpX_T.transpose(0,3,1,2)
+
+
+        res_dic = {}
+        res_dic['Ham'] = Ham_res/nrot
+        for X in self.matrix_list:
+            if self.matrix_bool[X]:
+                X_res = X+'_res'
+                res_dic[X] = vars()[X_res]/nrot
+        print('number of symmetry oprator == ',nrot)
+        if keep_New_R:
+                return res_dic , tmp_R_list
+        else:
+                return res_dic			
+
+    def symmetrize(self):
+        #====Time Reversal====
+        #syl: (sigma_y)^T *1j, syr: sigma_y*1j
+        if self.spin:
+            base_m = np.eye(self.num_wann//2)
+            syl=np.array([[0.0,-1.0],[1.0,0.0]])
+            syr=np.array([[0.0,1.0],[-1.0,0.0]])
+            self.ul=np.kron(syl,base_m)
+            self.ur=np.kron(syr,base_m)
+        
+        #========================================================
+        #symmetrize exist R vectors and find additional R vectors 
+        #========================================================
+        print('##########################')
+        print('Existing Block')
+        res_dic_1, iRvec_add =  self.average_H(self.iRvec,keep_New_R=True)
```

Comment:
one week point about that. In theory, after you find new vectors, if you repeat the procedure, you may find even more Rvectors (the new Rs from previous iteration transformed), until the procedure converges. 

What I would do, is to find All new Rvectors iteratively, then extend all amtrices to the new size (by zeros), and then perform the symmetrization.

Reply:
Already canceled the Existing Block.